### ISSUE TITLE
"default-more-channels" Not Working with DTS

### ISSUE THREAD

**Issue by Dead-Rekon** [mdhiggins/sickbeard_mp4_automator#1345]
I'm converting my library from mp4 to mkv, and defaulting all audio tracks to those with the most channels (mostly DTS/HD MA). A file with 5.1 AC3 and default 2.0 AAC will properly default to 5.1 AC3 upon conversion. However, it seems any form of DTS will not. All of my files have the first audio stream as AAC 2.0 default and secondary streams after, which is what I'm trying to reverse.

```
[Audio]
codec = aac,ac3,eac3,flac,mp3,dts
languages = eng,jpn
default-language = eng
first-stream-of-language = False
allow-language-relax = True
channel-bitrate = 0
max-bitrate = 0
max-channels = 0
prefer-more-channels = True
default-more-channels = True
filter = 
force-filter = False
sample-rates = 
copy-original = True
aac-adtstoasc = False
ignore-truehd = mp4, m4v
ignored-dispositions = 
unique-dispositions = False

[Universal Audio]
codec = 
channel-bitrate = 0
first-stream-only = False
move-after = False
filter = 
force-filter = False
```

**Comment by mdhiggins** [comment]
Can you post a log of one that's not working? And maybe an ffprobe of the resulting file

**Comment by mdhiggins** [comment]
```
    def sortStreams(self, streams, languages):
        if self.settings.sort_streams:
            self.log.debug("Reordering streams to be in accordance with approved languages and channels [sort-streams, prefer-more-channels].")
            if len(streams) > 0:
                if isinstance(streams[0], dict):
                    streams.sort(key=lambda x: x.get('channels', 999), reverse=self.settings.prefer_more_channels)
                    if languages:
                        streams.sort(key=lambda x: languages.index(x.get('language')) if x.get('language') in languages else 999)
                else:
                    streams.sort(key=lambda x: x.audio_channels, reverse=self.settings.prefer_more_channels)
                    if languages:
                        streams.sort(key=lambda x: languages.index(x.metadata.get('language')) if x.metadata.get('language') in languages else 999)
                    streams.sort(key=lambda x: x.disposition.get('comment'))
```

The stream sorter does not take codecs into account so this behavior is unexpected. Are the DTS containing source files already MKVs by chance?

**Comment by Dead-Rekon** [comment]
All of the files are in MP4 format. As far as I can tell, it's only an issue with DTS audio tracks for some reason. The one AC3 DD file I had properly set the 5.1 DD channel as default.

[Post-convert.txt](https://github.com/mdhiggins/sickbeard_mp4_automator/files/5365193/Post-convert.txt)
[Pre-convert.txt](https://github.com/mdhiggins/sickbeard_mp4_automator/files/5365194/Pre-convert.txt)

**Comment by mdhiggins** [comment]
Can you post a conversion log too?

**Comment by mdhiggins** [comment]
Preferably with DEBUG level logging set in logging.ini

```
[handler_fileHandler]
class = handlers.RotatingFileHandler
level = DEBUG
formatter = simpleFormatter
args = ('%(logfilename)s', 'a', 100000, 3)
```

**Comment by Dead-Rekon** [comment]
Regular and DEBUG in here.

[Convert Log.txt](https://github.com/mdhiggins/sickbeard_mp4_automator/files/5365488/Convert.Log.txt)

**Comment by mdhiggins** [comment]
The logs appear to be cut off and missing some important stuff, you cut off the input file JSON information and the debug logging only includes the FFMPEG progress which is relatively useless. I need the full debug log from start to finish for this file.

Because of this my interpretation is limited but it looks like this conversion has 2 source audio streams which is different from what you sent in the pre-convert.txt and I think that's part of the issue. My guess is that your source file has something marked as default that's being carried through but the logs are incomplete

**Comment by Dead-Rekon** [comment]
Hmmm, copied everything that printed. Here's both again from the original log file.

[sma.log](https://github.com/mdhiggins/sickbeard_mp4_automator/files/5365558/sma.log)

**Comment by mdhiggins** [comment]
That's much more helpful

The problem is here that your source file has the AAC track already set to default so its respecting the source material

Nothing to do with it being DTS

You can use the `sanitize-disposition` option to clear that information if you don't want to respect the source dispositions 

Just set `sanitize-disposition = default`

**Comment by Dead-Rekon** [comment]
That seemed to have solved the issue. I had set everything to 2.0 default because Plex Media Player and Roku Plex was smart enough to select the DTS tracks in combination with my receiver. However, Plex for Android will only select the default audio track so I had to manually change each track (12 remote clicks) every time a new movie or TV episode would play. Thanks a bunch. You've got the best support out there!
